### PR TITLE
[apr-util] Add crypto feature and don't require OpenSSL if not needed

### PR DESCRIPTION
**Pull request by vicroms**
Makes linking OpenSSL be guarded by the crypto feature reducing build time.

OpenSSL is only required when `APU_HAVE_CRYPTO` is enabled. This option is disabled by default, but the corresponding `find_package(OpenSSL)` is not guarded by the option. This PR disables `find_package(OpenSSL)` when it's not needed. As a bonus, users can now build with crypto support.

### CODE REVIEW

File: ports/apr-util/openssl-only-with-crypto-feature.patch [hunk 1]
```diff
@@ -0,0 +1,21 @@
+﻿diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 15526d6..7263743 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
```
Comment (BillyONeal) [inline ports/apr-util/openssl-only-with-crypto-feature.patch:4]: Have you submitted this to upstream?

File: ports/apr-util/openssl-only-with-crypto-feature.patch [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+-FIND_PACKAGE(OpenSSL)
+-
+ OPTION(APU_HAVE_CRYPTO      "Crypto support"                            OFF)
+ OPTION(APU_HAVE_ODBC        "Build ODBC DBD driver"                     ON)
+ OPTION(APR_HAS_LDAP         "LDAP support"                              ON)
+@@ -40,6 +38,7 @@ FOREACH(onelib ${APR_LIBRARIES})
+ ENDFOREACH()
+ 
+ IF(APU_HAVE_CRYPTO)
++FIND_PACKAGE(OpenSSL REQUIRED)
```
Comment (dg0yt) [inline ports/apr-util/openssl-only-with-crypto-feature.patch:18]: Isn't it enough to also  bind `CMAKE_REQUIRE_FIND_PACKAGE_OpenSSL` to feature crypto? (Are there more uses of 'OpennSSL_FOUND`?

File: ports/apr-util/openssl-only-with-crypto-feature.patch (hunk 1 above)
Comment (vicroms) [reply ports/apr-util/openssl-only-with-crypto-feature.patch]: Patch no longer needed.

File: ports/apr-util/portfile.cmake [hunk 3]
```diff
@@ -4,6 +4,15 @@ vcpkg_download_distfile(ARCHIVE
     SHA512 40eff8a37c0634f7fdddd6ca5e596b38de15fd10767a34c30bbe49c632816e8f3e1e230678034f578dd5816a94f246fb5dfdf48d644829af13bf28de3225205d
 )
 
+
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+FEATURES
+    crypto APU_HAVE_CRYPTO
+    crypto CMAKE_REQUIRE_FIND_PACKAGE_OpenSSL
+INVERTED_FEATURES
+    crypto CMAKE_DISABLE_FIND_PACKAGE_OpenSSL
```
Comment (dg0yt) [inline ports/apr-util/portfile.cmake:13]: FTR in general, this is more dangerous than the other option. It must only be used if there is no transitive dependency on OpenSSL in any other cmake package used by apr-util.

**Review by Adela0814 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**